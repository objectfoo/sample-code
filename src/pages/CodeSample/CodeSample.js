import React from 'react';
import injectStyle from 'react-jss';
import enhanceWithRoot from '../../common/enhanceWithRoot';

export const CodeSample = ({classes}) => {
  return (
    <main className={classes.root}>
      <h1 className={classes.headline}>Hi</h1>
    </main>
  );
};

// istanbul ignore next line
const styles = theme => ({
  root: {
    padding: theme.spacing.unit * 2,
    color: theme.palette.secondary
  },
  headline: {
    margin: 0
  }
});

export default enhanceWithRoot(
  injectStyle(styles)(CodeSample)
);
