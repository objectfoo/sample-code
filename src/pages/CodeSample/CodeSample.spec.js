import React from 'react';
import {shallow} from 'enzyme';
import {CodeSample} from './CodeSample';

describe('CodeSample', () => {
  it('should render', () => {
    const wrapper = shallow(<CodeSample classes={{
      headline: 'headline',
      root: 'root'
    }} />);

    expect(wrapper).toMatchSnapshot();
  });
});
