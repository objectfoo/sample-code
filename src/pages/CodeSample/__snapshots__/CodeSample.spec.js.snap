// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeSample should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={
        Object {
            "headline": "headline",
            "root": "root",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1
        className="headline"
>
        Hi
</h1>,
      "className": "root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hi",
        "className": "headline",
      },
      "ref": null,
      "rendered": "Hi",
      "type": "h1",
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1
          className="headline"
>
          Hi
</h1>,
        "className": "root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hi",
          "className": "headline",
        },
        "ref": null,
        "rendered": "Hi",
        "type": "h1",
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
